### DISCUSSION TITLE
[Question] Redirect from parent to root

### DISCUSSION THREAD

**Discussion by princestha** [csells/go_router#234] [Q&A] [answered]
Hi, if I have a scenario something like this...

I have my routes with this structure:
>user
>> :uid
>> settings/me
>> notifications/me

So when a user first comes to /user -> I want this redirected to :uid (This is working, however, when I go back using the back button, it goes back to a blank page and shows the user widget. How can I avoid this and redirect to home when this path is hit?

Thank you

--- I am experimenting and I can see that I can get this done with the push function as it will just push it on the stack, but is there a way to use the go functionality and when going back after the go, to re-run the redirect function when its back to the user route path?
Please let me know if this doesn't make sense.

**Comment by csells** [comment, marked as answer]
@princestha You probably want something like this:

```dart
GoRoute(path: '/user', redirect: (_) => '/user/default_user_id'),
GoRoute(path: '/user/:uid', builder: (c,s) => UserDetailsScreen(uid: s.params['uid'])),
```

Does that make sense?

**Reply by princestha** [reply]
Hi @csells, 
Yep, makes sense. I actually went ahead and implemented that and it is good now. However, I have a concern or questions rather: 

So if the structure is 
>user
>> :uid
>> settings/me
>> notifications/me

The problem I was having was when the context.pop() was clicked after navigating to user/MyUserId.
What happened here is that it would go back to the user page but then it wouldn't redirect. It would just render the users page which had nothing.

After pressing back which calls the context.pop, it should redirect back to the user's route (eg user/uid), right? For some reason for me, it didn't run the redirect function again once the back was clicked. 

Hope that makes sense. Else, I will try to build out a sample for you to test. 

Thank you

**Reply by csells** [reply]
When you use `context.go()`, you're building a whole stack of pages, not just the top-most page. That's why the `/user` route example I gave you is a top-level route that forwards to `/user/:uid` as another top-level route. I think what you probably want is something like this:

```dart
GoRoute(path: '/user', redirect: (_) => '/user/default_user_id'),
GoRoute(
  path: '/user/:uid',
  builder: (c,s) => UserDetailsScreen(uid: s.params['uid']),
  routes: [
    GoRoute(path: 'settings', builder: (c,s) => UserSettingsScreen(uid: s.params['uid']),
    GoRoute(path: 'notifications', builder: (c,s) => UserNotificationsScreen(uid: s.params['uid']),
  ],
),
```

When you `go('/user')`, you'll be forwarded to `/user/default_user_id`.

When you `go('/user/some_user_id')`, you'll have a stack with a single page: `UserDetailsScreen`

When you `go('/user/some_user_id/settings')`, you'll have a stack of two pages: `UserDetailsScreen`, `UserSettingsScreen`

When you `go('/user/some_user_id/notifications')`, you'll have a stack of two pages: `UserDetailsScreen`, `UserNotificationsScreen`

Does that make sense? Is that what you're after?

**Reply by princestha** [reply]
Hi @csells  yes that makes sense. Thank you so much for your detailed explanations.
Have a great day!
